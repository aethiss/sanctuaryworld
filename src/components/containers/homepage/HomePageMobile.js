import React from 'react'
import PropTypes from 'prop-types'

// Components
import DiabloButton from '../../commons/buttons/diabloButton/DiabloButton'

const styles = {
  container: {
    height: '100vh',
    width: '100%',
    padding: '5%',
    backgroundImage: 'url("/images/background/homepage-background-mobile.jpg")',
    backgroundSize: 'contain',
    backgroundRepeat: 'no-repeat',
    display: 'flex',
    flexDirection: 'column',
    justifyContent: 'flex-end',
    alignItems: 'center',
  },
  buttonContainer: {
    display: 'flex',
    marginBottom: '80%',
  },
  loginContainer: {
    display: 'flex',
    marginBottom: '80%',
  }
}

const loginRender = (battletag) => (
  <div style={styles.loginContainer}>
    <div>
      Connected as {battletag}
    </div>
  </div>
)

const connectButton = () => (
  <div style={styles.buttonContainer}>
    <DiabloButton text='Connect' anchorLink='/account' />
  </div>
)

const HomePageMobile = ({ battletag }) => {
  return (
    <div style={styles.container}>
      { battletag ? loginRender(battletag) : connectButton() }
    </div>
  )
}

HomePageMobile.propTypes = {
  battletag: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.bool,
  ])
}

HomePageMobile.defaultProps = {
  battletag: false,
}

export default HomePageMobile
