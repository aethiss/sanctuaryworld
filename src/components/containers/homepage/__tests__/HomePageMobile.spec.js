import React from 'react'
import HomePageMobile from '../HomePageMobile'
import TestRenderer from 'react-test-renderer'

describe('DiabloButton Component', () => {
  describe('Render', () => {
    it('[HomePageMobile] Default render', () => {
      const testRenderer = TestRenderer.create(<HomePageMobile />)
      expect(testRenderer).toMatchSnapshot()
    })
  })
})
