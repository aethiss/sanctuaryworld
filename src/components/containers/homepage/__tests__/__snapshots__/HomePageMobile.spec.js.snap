// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DiabloButton Component Render [HomePageMobile] Default render 1`] = `
<div
  style={
    Object {
      "alignItems": "center",
      "backgroundImage": "url(\\"/images/background/homepage-background-mobile.jpg\\")",
      "backgroundRepeat": "no-repeat",
      "backgroundSize": "contain",
      "display": "flex",
      "flexDirection": "column",
      "height": "100vh",
      "justifyContent": "flex-end",
      "padding": "5%",
      "width": "100%",
    }
  }
>
  <div
    style={
      Object {
        "display": "flex",
        "marginBottom": "80%",
      }
    }
  >
    <a
      href="/api/connect/bnet"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      <button
        className="makeStyles-base-1"
        onClick={[Function]}
      >
        Connect
      </button>
    </a>
  </div>
</div>
`;
